common/app/concurrent/FutureSemaphore.scala

```diff
@@ -0,0 +1,55 @@
+package concurrent
+
+import java.util.concurrent.TimeUnit
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import akka.util.Timeout
+import akka.pattern.{PipeToSupport, ask}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
+
+class FutureSemaphore(maxOperations: Int)(implicit ec: ExecutionContext) extends Actor with PipeToSupport {
```

Comment:
Did you consider using https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Semaphore.html and if so what were the reasons against?